main.js

```diff
@@ -78,13 +78,29 @@ const getAppContainer = (options) => {
 	/** @type {Promise<any>[]} */
 	const initPromises = [];
 
-	const instrumentListen = new InstrumentListen(express(), meta, initPromises);
+	const instrumentListen = new InstrumentListen(express(), meta, initPromises, options);
 	const app = instrumentListen.app;
 
 	const addInitPromise = initPromises.push.bind(initPromises);
 
-	// must be the first middleware
-	app.use(raven.requestHandler());
+	// Raven must be the first middleware if it's loaded
+	if (options.withSentry) {
+		// Note: we require n-raven here because importing n-raven introduces
+		// a lot of side effects. If we don't import it inside this conditional
+		// then it'll always set up unhandled rejection errors. This has a
+		// negligible impact on startup speed – the module has to be loaded
+		// synchronously regardless of whether it's in this conditional or not,
+		// we're just deferring it until later on, when the main `express`
+		// function is called
+		const raven = require('@financial-times/n-raven');
+		app.use(raven.requestHandler());
+	} else {
+		nLogger.warn({
+			event: 'EXPERIMENTAL_OPTION',
+			message: `The 'withSentry' option is experimental and should only be set to 'false' in non-critical applications for now. This may impact our ability to debug issues with this app.`,
+			optionName: 'withSentry'
+		});
```

Comment:
Just so I've got somewhere to put it for now, here's the future Splunk search to find any apps which have disabled Sentry:

```
index=heroku (source="dotcom-*" OR source="next-*" OR source="cp-*") event=EXPERIMENTAL_OPTION optionName=withSentry
```